python/pylibraft/pylibraft/neighbors/ivf_pq/ivf_pq.pyx

```diff
@@ -465,16 +474,23 @@ def extend(Index index, new_vectors, new_indices, handle=None):
         handle = Handle()
     cdef handle_t* handle_ = <handle_t*><size_t>handle.getHandle()
 
-    vecs_cai = cai_wrapper(new_vectors)
+    try:
+        vecs_cai = cai_wrapper(new_vectors)
```

Comment:
We don't need to do so in this PR but I wonder if we should eventually encapsulate this inside a simple factory function that dispatches to the correct wrapper based on whether it's host or device? I can just see this pattern getting repetitive if we end up using it in more algorithms in the future.

Reply:
I have added a factory function.